lib/Interpreter/IncrementalJIT.cpp

```diff
@@ -217,13 +218,19 @@ IncrementalJIT::getSymbolAddressWithoutMangling(llvm::StringRef Name,
   if (auto Sym = getInjectedSymbols(Name))
     return Sym;
 
+  const std::string SymName = Name;
```

Comment:
`llvm::StringRef::data()` does not guarantee that the content is null terminated and if the StringRef is referring to a sub-part of the original input, data() will not have an null character at the 'end' of the StringRef.   So data() can not be used when a c-string is expected.